### ISSUE TITLE
Match UDID to authentication

### ISSUE THREAD

**Issue by jessepeterson** [micromdm/micromdm#358]
Right now we trust that a device gives us its correct UDID in the command (connect) and checkin endpoints. However in this way a compromised device cert and/or rogue device (however unlikely) is able to read any given device's command queue by simply forging another UDID.

To consider implementing a check in those endpoints (or some middleware, etc.) that checks that a provided UDID matches the authentication (client certificate/cert. signature) given to us in the request.

**Comment by groob** [comment]
Agreed. I think this is also a recommended best practice in the spec! I remember seeing something related to this recently.

**Comment by groob** [comment]
<img width="1500" alt="screenshot 2017-12-12 21 16 02" src="https://user-images.githubusercontent.com/1526945/33918498-afea16d2-df81-11e7-8bb6-8726e8509f31.png">

**Comment by jessepeterson** [comment]
I was thinking about how to do this. I think we manage both serial number and UDID keys in the device DB by reference. Perhaps we add a key into the device DB by auth certificate fingerprint?

**Comment by groob** [comment]
I think this could be implemented similar to how the device block/unblock is implemented as a connect service middleware. 

```
func AuthCertMiddleware(store Store) connect.Middleware {
	return func(next connect.Service) connect.Service {
		return &removeMiddleware{
			store: store,
			next:  next,
		}
	}
}
```

```
func (mw authMw) Acknowledge(ctx context.Context, req connect.MDMConnectRequest) ([]byte, error) { 
// use the db method here to authenticate the incoming request. 
}
```

https://github.com/micromdm/micromdm/blob/6e561224d0311a23a648ccc0f9e351e4cce6d7d5/mdm/connect/middleware.go#L10
